### ISSUE TITLE
As a user, I want the WRES to obtain ensemble forecasts from the WRDS HEFS service in the cloud

### ISSUE THREAD

**Issue by HankHerr-NOAA** [NOAA-OWP/wres#53]
Putting this on our radar. The WRDS HEFS service in the Cloud is still in development, but has been deployed to their development-Cloud and is slated to be deployed by the end of this FY. I'm told it serves 30 days of data now, and hopefully 90 days in the future. I think this is enough that the WRES should be able to pull data from it, especially since we already have folks doing evaluations that are that short.  

Details about the web service URL and usage are TBD. And note that we have no RFC or other user currently asking for this, though it has been mentioned in the distant past.  

Thanks,

Hank

**Comment by james-d-brown** [comment]
I would expect/hope that there is literally nothing to do. They should not be changing the apis, for example. Perhaps we need to bundle a new cert, but hopefully not.

Anyway, there is probably relatively little incentive for users to leverage this instance over the NWCAL instance. If they removed the NWCAL instance or WRES was put in the cloud alongside it in a way that made evaluations more performant, yes, but I don't believe either of those two scenarios is impending.

**Comment by james-d-brown** [comment]
But I guess this could be a "1 banana" ticket to check that it works.

**Comment by HankHerr-NOAA** [comment]
I won't be that easy. They are moving to a more generalized framework for their APIs in the Cloud that makes use of GraphQL. I'm almost certain that will lead to a change in JSON schema. My hope is that moving forward any new service or existing service when moved the Cloud will follow the new HEFS JSON schema (assuming it is new), given ensemble data is as complicated as it gets. (Or, for existing services, the JSON will remain unchanged.)

In other words, this will likely be quite a few "bananas".

Hank

**Comment by james-d-brown** [comment]
Sigh, this is like the fifteenth API from WRDS.

OK, well we'll probably need to read from the new API at some point, but I would definitely put that work before this work (well, that is unavoidable, but it shouldn't be rolled up with this ticket). The HEFS service is probably the first case where it may be needed.

**Comment by james-d-brown** [comment]
Yes, I can access this.

**Comment by james-d-brown** [comment]
Bottom line, I can work on this one in due course.

**Comment by HankHerr-NOAA** [comment]
Thanks for confirming. I'm told that, once the ingester is fixed (its down at the moment so data is old), they will serve 30 days of HEFS. They'll then reevaluate it later to see if that costs too much or if more can be served.

Still not 100% sure that site is "ready for primetime", but its close. The next WRES/WRDS is this Thursday, so I'll ask.

Thanks,

Hank

**Comment by HankHerr-NOAA** [comment]
I believe this is redundant with #138 and should be closed. Agree?

Hank

**Comment by james-d-brown** [comment]
Yes, I don't think there's a way to mark tickets duplicate.

**Comment by HankHerr-NOAA** [comment]
Found it...

**Comment by HankHerr-NOAA** [comment]
Nope. Didn't work. Trying again...
